### ISSUE TITLE
Support downloading and removing DRM from Kindle Digital ebooks (not visible on Audible)

### ISSUE THREAD

**Issue by shirakaba** [rmcrackan/Libation#813]
**Is your feature request related to a problem? Please describe.**

Although evidently Libation is capable of downloading and de-DRMing [Audible books](https://github.com/rmcrackan/Libation/blob/master/Documentation/GettingStarted.md#download-your-books----drm-free), I am unable to access the ebooks for Kindle that I have purchased on Amazon Japan. This is despite my using the same account on both [amazon.co.jp](https://www.amazon.co.jp) and [audible.co.jp](https://www.audible.co.jp). They simply do not appear in my library (while the Audible audiobooks do).

An example is the light novel for "Spice and Wolf" volume 1 [on amazon.co.jp](https://www.amazon.co.jp/-/en/gp/product/B00GOO636G), as well as its corresponding audibook [on audible.co.jp](https://www.audible.co.jp/pd/1巻-狼と香辛料-オーディオブック/B0C58LD5WG). Although there is clearly metadata connecting the two across websites, despite me having purchased both, only the audiobook shows up in Libation.

It may be a problem specific to Amazon Japan, because they're an unusual entity.

**Describe the solution you'd like**
It would be great if Libation could download and de-DRM these ebooks as well. I am not sure whether the API for downloading Kindle ebooks has been reverse-engineered by anyone (and it may well depend upon having a physical Kindle registered, but that requirement is acceptable), but ideally the app would download the ebook and de-DRM it.

The state-of-the-art for deDRM seems to be the [DeDRM](https://apprenticealf.wordpress.com/2012/09/10/calibre-plugins-the-simplest-option-for-removing-most-ebook-drm/) plugin for [Calibre](https://calibre-ebook.com).

**Describe alternatives you've considered**
Right now, the only option is to use an alternative workflow like downloading the ebook to a physical Kindle device, then connecting it to the computer, and doing the rest with Calibre and DeDRM, which is very fiddly. It would be incredible to have this all housed in a single app, perhaps with a basis on something like [docker-kindle](https://github.com/igjit/docker-kindle) or figuring out the API for downloading a file from Amazon via "Manage Your Content and Devices" on the Amazon shopping site.

**Comment by rmcrackan** [comment]
Libation is for audible audiobooks only. I've toyed with the idea of other audiobook services and/or things like kindle but ultimately decided against it.

I'm clearly a big fan of owning the digital things I buy and I think DRM is a scourge. Although it won't be from me, I do very much hope someone else takes up the challenge for these other services and products like kindle.

**Comment by mkb79** [comment]
@shirakaba 
Some years ago I wrote a [Python package](https://github.com/mkb79/kindle) which can download and decrypt Kindle ebooks. Maybe this works today?!

**Comment by rmcrackan** [comment]
@mkb79 My first reaction: I didn't know he wrote a kindle de-drm. My immediate second reaction: of course he did; he's just awesome like that :)

**Comment by mkb79** [comment]
As I wrote the app I had a lot of Kindle ebooks. And it was always very tedious to download and decipher them. And the app worked fine for me. But I last used it 1 or 2 years ago. Therefore, I do not know whether the API and encryption are still unchanged.

**Comment by shirakaba** [comment]
Oh wow! Well if the de-DRM implementation is based on apprenticeharper's [implementation](https://github.com/apprenticeharper/DeDRM_tools) [as you say](https://github.com/mkb79/kindle/blob/98c06a7073082cb0516a9359952cd1c4e9a26cdb/src/kindle/dedrm.py#L2C26-L2C64), that hasn't changed since April 2021, so should still work as long as Amazon still serves the files in the same old format from the original API endpoints.

I'll have a look over the code!

**Comment by mkb79** [comment]
@shirakaba 
Yes, I've used the code as a base and rewrote it to use the `amazon.ion` package. It would be great if you can report back. Thanks.

**Comment by rmcrackan** [comment]
tl;dr: sorry, nope

Longer answer: it's not a simple thing. The entirety of Libation has co-evolved with audible. There are hundreds of tiny choices which are all audible-centric. The idioms in right-click menus, settings for what to do with subscriptions and podcasts, workflows for how to deal with aax vs mp3 files, the audio libraries for chapter splitting and ID3 tag management, and even how the database is structured around audible book ID numbers. Kindle books just wouldn't be a good fit for Libation.

**Comment by shirakaba** [comment]
Fair enough! Totally understand.

**Comment by CLHatch** [comment]
I personally use some deDRM plugins for Calibre to download my Kindle books.  Rather cumbersome (I really dislike Calibre), but this is the plugin package I use: https://github.com/noDRM/DeDRM_tools

It ends up in a rather useless folder structure, because the dev doesn't believe in using folder structures, and organizes everything in his internal database instead, but at least you can then do a "Save to disk" to export it into a useful format.  Only caveat is the Calibre dev is also an idiot that doesn't think he should allow long pathnames in Windows, with the reasoning that "other apps might not use long paths, so it will break for them".  Ah, the irony of a dev being part of the problem, and saying "it's because these other devs apps will cause a problem".

Anyways, this makes "Save to disk" save out as `Author1; Author2; etc/Series/## - (YEAR) - Book Title with Subtitle [ASIN]/Book Title [ASIN].epub` (skipping the `Series/## - ` part if it's not in a series):
`{authors:re( & ,; )}/{series:re(:,꞉)}/{series_index:0>2s|| - }({pubdate:format_date(yyyy)}) - {title:re(:,꞉)} [{identifiers:select(mobi-asin)}]/{title:sublist(0,1,:)} [{identifiers:select(mobi-asin)}]`
